A55-stateful-session-affinity.md

```diff
@@ -0,0 +1,398 @@
+           cookie:
+             name: global-session-cookie
+             path: /Package1.Service2/Method3
+             ttl: 120s
+```
+
+`name` is the name of the cookie used in this feature.
+
+`path` is the path for which this cookie is applied. **Note:** the path value
+here needs to match any path specified in a RouteConfiguration when route
+based [Filter Config Override][filter_config_override] is used.
+
+
+`ttl` is the time-to-live for the cookie. It is set on the cookie but will not
+be enforced by gRPC.
+
+This new filter will be added to the HTTP Filter Registry and processed as
+described in [A39: xDS HTTP Filter Support][A39]. Specifically when this filter
+is present in the configuration, gRPC will create the appropriate client filter
+(aka interceptor in Java/Go) and install it in the channel to process data
+plane RPCs. We call this filter `CookieBasedStatefulSessionFilter`. It will
+copy the 3 configuration values - `name`, `ttl`  and `path` -  into the filter
+object. Validation and processing are further described in
+[Stateful session][stateful-session].
+
+Note that
+[StatefulSessionPerRoute][stateful-session-per-route] *will* be supported as
+[Filter Config Overrides][filter_config_override]. For example, let's say
+the top level config (in `HttpConnectionManager.http_filters`) contains certain
+`StatefulSession` configuration. This configuration can either be
+overridden or disabled by a
+[StatefulSessionPerRoute][stateful-session-per-route] config for a particular
+virtual host or route through a merged configuration as described in
+[A39 Filter Config Overrides][filter_config_override].
+
+The filter is described in more detail in the section
+[`CookieBasedStatefulSessionFilter` Processing][filter-section].
+
+### Load Balancer Configuration Containing `override_host_status`
+
+Even with stateful session affinity enabled, gRPC will only send an RPC to a
+valid backend (host) as defined by
+[`common_lb_config.override_host_status`][or-host-status]. When the backend
+state is invalid, gRPC will use the configured load balancing policy to pick
+the backend.
+
+Note that gRPC currently ignores endpoints with health status other than
+`HEALTHY` or `UNKNOWN` and even `XdsClient` excludes such endpoints in its
+update to the watchers. We will need a separate design to include such
+endpoints if we determine that such behavior is needed for this feature to
+be useful to our users.
+
+As a result, for this design gRPC will only consider the `UNKNOWN` or
+`HEALTHY` states that are specified in
+[`common_lb_config.override_host_status`][or-host-status] and ignore all
+other values. We considered the alternative of NACKing (i.e. rejecting)
+the CDS update when unsupported values are present, however it was discarded
+because of the difficulty in using such configuration in mixed deployments.
+
+The `override_host_status` value is included in the config update sent to
```

Comment:
If we are actually treating `UNKNOWN` and `HEALTHY` the same, then we probably don't need to plumb the original `override_host_status` value in the `xds_cluster_resolver` LB policy config; instead, we just need to plumb down a boolean indicating whether to configure the host-override functionality.

Reply:
Good catch. Fixed.

Reply:
fixed